### DISCUSSION TITLE
DB::transaction function should revert files as well.

### DISCUSSION THREAD

**Discussion by Dasein-hub** [laravel/framework#40437] [Ideas]
I've recently discovered that DB::transaction function doesn't revert file changes like renaming, replacing, etc. Would be nice if it did.

**Comment by dennisprudlo** [comment]
Transactions are a feature of your database not of Laravel and your database doesn't know what files you have.

You will have to track the changes you did by yourself and then revert them.

**Reply by Dasein-hub** [reply]
Then, the Storage class should have some transaction like behaviour. Like, if it's in try catch and exception fires, it should revert the changes.
